faunadb/client.py

```diff
@@ -139,98 +130,37 @@ def ping(self, scope=None, timeout=None):
     """
     return self.get("ping", {"scope": scope, "timeout": timeout})
 
-  def _log(self, indented, logged):
-    """Indents `logged` before sending it to self.logger."""
-    if indented:
-      indent_str = "  "
-      logged = indent_str + ("\n" + indent_str).join(logged.split("\n"))
-
-    if _debug_logger:
-      _debug_logger.debug(logged)
-    if self.logger:
-      self.logger.debug(logged)
-
   def _execute(self, action, path, data=None, query=None):
     """Performs an HTTP action, logs it, and looks for errors."""
-    # pylint: disable=too-many-branches
-
     if isinstance(path, Ref):
       path = path.value
     if query is not None:
       query = {k: v for k, v in query.iteritems() if v is not None}
 
-    if self.logger is not None or _debug_logger is not None:
-      self._log(False, "Fauna %s /%s%s" % (action, path, Client._query_string_for_logging(query)))
-
-      if self.session.auth is None:
-        self._log(True, "Credentials: None")
-      else:
-        self._log(True, "Credentials: %s:%s" % self.session.auth)
-
-      if data:
-        self._log(True, "Request JSON: %s" % to_json(data, pretty=True))
-
-      real_time_begin = time()
-      response = self._execute_without_logging(action, path, data, query)
-      real_time_end = time()
-      real_time = real_time_end - real_time_begin
-
-      # response.headers a CaseInsensitiveDict, which can't be converted to JSON directly
-      headers_json = to_json(dict(response.headers), pretty=True)
-      response_dict = parse_json(response.text)
-      response_json = to_json(response_dict, pretty=True)
-      self._log(True, "Response headers: %s" % headers_json)
-      self._log(True, "Response JSON: %s" % response_json)
-      self._log(
-        True,
-        "Response (%i): API processing %sms, network latency %ims" % (
-          response.status_code,
-          response.headers.get("X-HTTP-Request-Processing-Time", "N/A"),
-          int(real_time * 1000)))
-      return Client._handle_response(response, response_dict)
+    start_time = time()
+    response = self._perform_request(action, path, data, query)
+    end_time = time()
+    response_dict = parse_json(response.text)
+    if self.observer is not None:
+      self.observer(RequestResponseInfo(
+        self,
+        RequestInfo(action, path, query, data),
+        # response.headers is a CaseInsensitiveDict, we don't need that
```

Comment:
All CaseInsensitiveDict is make using the key "X-SomethingOrOther" the same as "x-somethingorother" (etc), so you don't have to have the exact case, as http headers aren't case sensitive. I don't think there's really any harm in exposing this as-is.